pages/plant/[id].js

```diff
@@ -92,11 +111,29 @@ export default function PlantDetails({ plants, onDeletePlant }) {
       setToggleButtonName("Delete");
     }
   }
-
+  
+  function handleEdit(updatedPlant) {
+    onEditPlant(plantData.id, updatedPlant);
+  }
   return (
     <>
       <BackLink />
 
+      
+      <StyledEditButton onClick={toggleFormVisibility}>
+        <Image
+              src={"/icons/pencil-solid.svg"}
+              width={25}
+              height={25}
+              alt="Icon of a dead plant"
+              unoptimized
+            />
+      </StyledEditButton>
+      {isFormVisible && <>
+      <PlantForm onCreatePlant={onCreatePlant} onEditPlant={handleEdit} isEditMode={true} initialData={plantData}/>
+      </>
```

Comment:
```js
           <PlantForm
          onCreatePlant={onCreatePlant}
          onEditPlant={handleEdit}
          // pass `onCancel` as a prop to allow the form to close itself
          onCancel={handleCancel}
          // consider using `isFormVisible` here for edit control instead of `isEditMode` prop 
          isFormVisible
          initialData={plantData}
        />
```